ui-components/src/page_components/listing/ListingMap.tsx

```diff
@@ -35,6 +36,11 @@ const isValidLongitude = (longitude: number) => {
 }
 
 const ListingMap = (props: ListingMapProps) => {
+  // Lazy load the map component only when it's visible on screen
+  const { setIntersectingElement, intersecting } = useIntersect({ rootMargin: "500px" })
```

Comment:
Can you explain why it's exactly `500px`? Is there any option (and sense) to use e.g ` window.innerHeight` value?

Reply:
Oh I like that idea…500px was just an arbitrary value I chose so when someone's leisurely scrolling along the page they probably wouldn't see MapBox start up right in view.